packages/utils/src/graphql.ts

```diff
@@ -0,0 +1,49 @@
+interface GraphQLOperation {
+  operationType: string | undefined;
+  operationName: string | undefined;
+}
+
+/**
+ * Extract the name and type of the operation from the GraphQL query.
+ * @param query
+ */
+export function parseGraphQLQuery(query: string): GraphQLOperation {
+  const queryRe = /^(?:\s*)(query|mutation|subscription)(?:\s*)(\w+)(?:\s*)[{(]/;
+
+  const matched = query.match(queryRe);
+
+  if (matched) {
+    return {
+      operationType: matched[1],
+      operationName: matched[2],
+    };
+  }
+  return {
+    operationType: undefined,
+    operationName: undefined,
+  };
+}
+
+/**
+ * Extract the payload of a request ONLY if it's GraphQL.
+ * @param payload - A valid JSON string
+ * @returns A POJO or undefined
+ */
+export function getGraphQLRequestPayload(payload: string): any | undefined {
+  let graphqlBody = undefined;
+  try {
+    const requestBody = JSON.parse(payload);
+
+    // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access
+    const isGraphQLRequest = !!requestBody['query'];
```

Comment:
This may not be the most accurate _and_ safest way to detect a graphql request?

Reply:
If we only move this into the graphql integration, instead of runing this for every request, we can IMHO safely make this check, because stuff already ran through the `endpoints` check, so we can assume it should be fine!